software/libdvi/dvi.c

```diff
@@ -316,6 +341,12 @@ void dvi_audio_init(struct dvi_inst *inst) {
 void dvi_enable_data_island(struct dvi_inst *inst) {
     inst->data_island_is_enabled  = true;
 
+    dvi_setup_scanline_for_vblank_with_audio(inst->timing, inst->dma_cfg, true, &inst->dma_list_vblank_sync);
```

Comment:
The DMA list is now re-initialized with the required configuration when enabling data islands.